roles/supybot/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+# Create and configure a Buildbot master to watch changes in Buildbot's git
+# repo.
+---
+- name: update supybot repository
+  sudo_user: "{{ supybot_user }}"
+  git:
+    repo: "{{ supybot_repo }}"
+    dest: "{{ supybot_user_home }}/supybot-repo"
+    update: yes
+    version: "{{supybot_branch}}"
+    accept_hostkey: True
+  register: supybot_repo_state
+
+- name: install supybot in virtualenv
+  # note that pip install -e appears not to work for this package; uninstall and re-install instead
+  shell: "{{ supybot_env_dir }}/bin/pip uninstall supybot; {{ supybot_env_dir }}/bin/pip install {{ supybot_user_home }}/supybot-repo"
+  when: "supybot_repo_state|changed"
+
+- name: update meetbot repository
+  sudo_user: "{{ supybot_user }}"
+  git:
+    repo: "{{ meetbot_repo }}"
+    dest: "{{ supybot_user_home }}/meetbot-repo"
+    update: yes
+    version: "{{meetbot_branch}}"
+    accept_hostkey: True
+  register: meetbot_repo_state
+
+- name: create directories
+  file:
+    dest: "{{ item }}"
+    state: directory
+    owner: "{{ supybot_user }}"
+  with_items:
+    - "{{ supybot_user_home }}/run"
+    - "{{ supybot_user_home }}/run/conf"
+    - "{{ supybot_user_home }}/run/data"
+    - "{{ supybot_user_home }}/run/data/tmp"
+    - "{{ supybot_user_home }}/run/logs"
+    - "{{ supybot_user_home }}/run/plugins"
+    - "{{ supybot_user_home }}/run/backup"
+
+- name: link meetbot into supybot
+  file:
+    dest: "{{ supybot_user_home }}/run/plugins/MeetBot"
+    src: "{{ supybot_user_home }}/meetbot-repo/MeetBot"
+    state: link
+
+- name: link ircmeeting into supybot virtualenv
+  file:
+    dest: "{{ supybot_env_dir }}/lib/python2.7/site-packages/ircmeeting"
+    src: "{{ supybot_user_home }}/meetbot-repo/ircmeeting"
+    state: link
+
+- name: install config
+  copy:
+    # note that supybot is not allowed to write its config
+    dest: "{{ supybot_user_home }}/run/bb-supy.conf"
+    src: bb-supy.conf
+
+- name: install meetbot config
+  copy:
+    dest: "{{ supybot_user_home }}/run/meetingLocalConfig.py"
+    src: meetingLocalConfig.py
+
+- name: set up dynamic config
+  file:
+    dest: "{{ supybot_user_home }}/conf/{{ item }}"
+    state: touch
+    owner: "{{ supybot_user }}"
+  with_items:
+    - channels.conf
+    - ignores.conf 
+    - userdata.conf
+    - users.conf
+
+# TODO: only when changed
```

Comment:
The `notify` action that gets triggered can do anything, including running a shell script or command. To make this only happen when something changes, find the steps that indicate something changed, have them notify `restart supybot` and move this step to `handlers/main.yml` with the same name.